jsonParser.go

```diff
@@ -130,39 +128,135 @@ type Link struct {
 	ContentVersion      string `json:"content-version"`
 	IntendedApplication string `json:"intended-application"`
 }
-
 type Primary struct {
 	URL string `json:"URL"`
 }
 type Resource struct {
 	Primary Primary `json:"primary"`
 }
-type Issued struct {
-	DateParts [][]int `json:"date-parts"`
-}
 type JournalIssue struct {
-	Issue           *string          `json:"issue"`
-	PublishedOnline *PublishedOnline `json:"published-online"`
-	PublishedPrint  *PublishedPrint  `json:"published-print"`
+	Issue           *string    `json:"issue"`
+	PublishedOnline *DateParts `json:"published-online"`
+	PublishedPrint  *DateParts `json:"published-print"`
 }
 type IssnType struct {
 	Value string `json:"value"`
 	Type  string `json:"type"`
 }
-type PublishedOther struct {
-	DateParts [][]int `json:"date-parts"`
-}
-type Published struct {
-	DateParts [][]int `json:"date-parts"`
-}
-
 type Relation struct {
 	Cites []any `json:"cites"`
 }
-
 type License struct {
 	URL            string  `json:"URL"`
 	Start          Indexed `json:"start"`
 	DelayInDays    int     `json:"delay-in-days"`
 	ContentVersion string  `json:"content-version"`
 }
+
+//
+func (pub *CrossRef) GetSimpleTitle() []string {
+	simpleTitle := pub.Title
+	if len(simpleTitle) == 0 {
+		return []string{""}
+	}
+	for pos, title := range simpleTitle {
+		title = strings.Replace(title, "\n", " ", -1)
+		title = strings.Replace(title, "( )+", " ", -1)
+		simpleTitle[pos] = strings.TrimSpace(title)
+	}
+	return simpleTitle
+}
+
+func (pub *CrossRef) GetSimpleDOI() string {
+	return pub.Doi
+}
+
+func (pub *CrossRef) GetSimpleAuthor() string {
+	if len(pub.Author) == 0 {
+		return ""
+	}
+	var simpleAuthor strings.Builder
+	for _, auth := range pub.Author {
+		simpleAuthor.WriteString(*auth.Family + " ")
+	}
+	return strings.TrimSpace(simpleAuthor.String())
```

Comment:
A simpler approach would be to add all the names to a slice of strings and then just do `strings.Join(authors, " ")`

Reply:
Also, should be a function

Reply:
I am not sure how it could be a simpler approach.

Reply:
A matter of preference I guess, it's the same code under the hood I'm pretty sure but you don't have to worry about trimming etc.

Also, just saw that this would stack multiple spaces in the case of empty family names